components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -452,6 +438,25 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
       state.externalWallet = payload.wallet
       break
     }
+    case types.ON_ANON_WALLET_STATUS: {
+      state = { ...state }
+      if (payload.result === RewardsExtension.Result.WALLET_CORRUPT) {
+        state.walletCorrupted = true
+        state.walletCreating = false
+        state.walletCreated = false
+        state.walletCreateFailed = false
+      } else if (payload.result === RewardsExtension.Result.WALLET_CREATED) {
+        state.walletCreated = true
+        state.walletCreateFailed = false
+        state.walletCreating = false
+        state.walletCorrupted = false
+      } else {
+        state.walletCreateFailed = false
+        state.walletCreating = false
+        state.walletCreated = false
+        state.walletCorrupted = false
```

Comment:
Stylistic choice - I'd suggest setting all of them to = false before the if / else if / else, and only setting the one's to true where needed. IMHO it would be more readable and easier to maintain.